assets/javascripts/discourse/initializers/extend-for-assigns.js

```diff
@@ -37,142 +38,52 @@ function defaultTitle(topic) {
   }
 }
 
-function includeIsAssignedOnTopic(api) {
+function extendTopicModel(api) {
   api.modifyClass("model:topic", {
     pluginId: PLUGIN_ID,
-    isAssigned() {
-      return this.assigned_to_user || this.assigned_to_group;
-    },
-  });
-}
 
-function registerTopicFooterButtons(api) {
-  registerTopicFooterDropdown({
-    id: "reassign",
+    assignees() {
+      const result = [];
 
-    async action(id) {
-      if (!this.currentUser?.can_assign) {
-        return;
+      if (this.assigned_to_user) {
+        result.push(this.assigned_to_user);
       }
 
-      const taskActions = getOwner(this).lookup("service:task-actions");
-
-      switch (id) {
-        case "unassign": {
-          this.set("topic.assigned_to_user", null);
-          this.set("topic.assigned_to_group", null);
-
-          await taskActions.unassign(this.topic.id);
-
-          this.appEvents.trigger("post-stream:refresh", {
-            id: this.topic.postStream.firstPostId,
-          });
-          break;
-        }
-        case "reassign-self": {
-          this.set("topic.assigned_to_user", null);
-          this.set("topic.assigned_to_group", null);
+      const postAssignees = this.assignedPosts().map((p) => p.assigned_to);
+      result.push(...postAssignees);
+      return result;
+    },
 
-          await taskActions.reassignUserToTopic(this.currentUser, this.topic);
+    uniqueAssignees() {
+      const map = new Map();
+      this.assignees().forEach((user) => map.set(user.username, user));
+      return [...map.values()];
```

Comment:
I _think_ this could be a one-liner?

```suggestion
      return [...new Set(this.assignees().map((u) => u.username))];
```

Reply:
Your version of code does another thing. We need to return an array of unique assignee objects here, while your one-liner returns an array of unique usernames.